api/fs/core.go

```diff
@@ -0,0 +1,56 @@
+// Package apifs is used by the Wash API to convert local files/directories
+// into Wash entries
+package apifs
+
+import (
+	"os"
+	"syscall"
+	"time"
+
+	"github.com/puppetlabs/wash/plugin"
+)
+
+// fsnode => filesystem node
+type fsnode struct {
+	plugin.EntryBase
+	path string
+}
+
+func newFSNode(finfo os.FileInfo, path string) *fsnode {
+	// TODO: Need to case this on platform since finfo.Sys()
+	// contains platform-specific data
+	statT := finfo.Sys().(*syscall.Stat_t)
+	attr := plugin.EntryAttributes{}
+	attr.
+		SetCtime(timespecToTime(statT.Ctimespec)).
+		SetAtime(timespecToTime(statT.Atimespec)).
+		SetMtime(timespecToTime(statT.Mtimespec)).
+		SetMode(os.FileMode(statT.Mode)).
+		SetSize(uint64(finfo.Size())).
+		SetMeta(plugin.ToMeta(finfo))
```

Comment:
There's some weird behavior going on with JSON serialization b/c `wash meta` outputs an empty hash for local files/dirs when it shouldn't. I'd like to address that in a separate PR though.